src/ILLink.RoslynAnalyzer/MSBuildPropertyOptionNames.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Linq;
+
+namespace ILLink.RoslynAnalyzer
+{
+	public static class MSBuildPropertyOptionNames
+	{
+		public const string PublishSingleFile = nameof (PublishSingleFile);
```

Comment:
We probably need to add `PublishTrimmed` now since otherwise all the trim warnings will automatically be turned on during single-file

Reply:
Wouldn't be better to think about the more general area than to hardcode the single narrow scenario? Are we going to add a property for each of remaining scenario which would benefit from the same logic?

Reply:
Perhaps there's some broader feature work we could do, but we certainly need to recognize whether trimming or single-file are enabled, so we at least need to recognize the two MSBuild properties that already exist.